### DISCUSSION TITLE
Too stupid to create a pull request... patch for adding additional sox spectrogram params

### DISCUSSION THREAD

**Discussion by hoover67** [mcguirepr89/BirdNET-Pi#413] [Ideas]
Hi folks,

apparently I'm too stupid to create a proper pull request (I tried and it generated a request based on an ancient version from somebody else instead of Patrick's fantastic work) so I'm attaching this git-generated patch to today's main/origin branch. It will allow you to change some parameters about how the sox sonagram is created (grayscale, width etc) using a new config variable in birdnet.conf.

Affected files: scripts/spectrogram.sh, scripts/extract_new_birdsounds.sh, birdnet.conf-defaults

I hope you find it useful! 

All the best, Uwe 
[patch_sox_spectrogram.zip](https://github.com/mcguirepr89/BirdNET-Pi/files/9009530/patch_sox_spectrogram.zip)

**Comment by nilspupils** [comment]
Great! I'm looking forward to seeing it!

**Reply by hoover67** [reply]
Thanks Nils, you can already make the changes locally if you want to, they're very really minor (a new variable is added to /etc/birdnet/birdnet.conf which is used in the two shell scripts outlined above). 

All the best,

Uwe

**Comment by mcguirepr89** [comment]
Thanks, @hoover67 -- I'll take a look through this diff and will put together a branch to test this out on -- we'll likely want to build out the ability to affect these settings somewhere in the Web Interface's "Tools" > "Settings" or (more likely) "Advanced Settings"

I will likely also add a variable that will allow users to adjust the speed at which the live spectrogram scrolls across the screen.

I'll update things for both features here as those changes are implemented and tested

My very best regards,
Patrick

**Reply by mcguirepr89** [reply]
- [ ] Add new variable to `install_config.sh`
- [ ] Add new variable to `update_birdnet_snippets.sh`
- [ ] Add `if . . . else` in `extract_new_birdsounds.sh` and `spectrogram.sh` in case the `sox` arguments are malformed causing the service to fail
- [ ] Add new variable adjustment/configuration in "Advanced Settings"

**Reply by hoover67** [reply]
Hey Patrick,

thanks much for your consideration! I was thinking about making the variable configurable from the GUI, but the "quick and dirty solution" was just too tempting to ignore 😸 

All the best, 

Uwe
